java/src/jmri/jmrix/AbstractSimulatorConnectionConfig.java

```diff
@@ -245,7 +245,7 @@ public void setManufacturer(String manufacturer) {
     }
 
     public String getConnectionName() {
-        if (adapter != null && adapter.getSystemConnectionMemo() != null) {
```

Comment:
I removed the protection against an NPE to confirm that things are working. At this point, I think we don't want to mask the NPE.